src/components/LoginForm/LoginForm.jsx

```diff
@@ -1,47 +1,58 @@
 import {
+  Alert,
+  AlertIcon,
   Button,
   FormControl,
   FormErrorMessage,
-  FormHelperText,
   FormLabel,
   Input,
   VStack,
 } from "@chakra-ui/react";
 import { validationMessages } from "constants";
 import { useFormik } from "formik";
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import { useNavigate } from "react-router-dom";
+import { loginUser } from "services";
 import * as Yup from "yup";
 
 const LoginForm = () => {
 
   const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const {authToken, authStatus} = useSelector((state)=>state.auth);
 
   const formik = useFormik({
     initialValues: {
-      email: "",
+      username: "",
       password: "",
     },
     validationSchema: Yup.object({
-      email: Yup.string()
-        .email(validationMessages.emailFormat)
+      username: Yup.string()
         .required(validationMessages.emailEmpty),
       password: Yup.string()
         .required(validationMessages.passwordEmpty)
         .min(8, validationMessages.passwordShort),
     }),
     onSubmit: (values, actions) => {
+      const {username, password} = values;
+      dispatch(loginUser({username, password}))
       actions.resetForm();
-      navigate("/feed", {replace: true});
     },
   });
 
+  useEffect(()=>{
+      if(authToken){
+        navigate("/feed", {replace: true});
+      }
```

Comment:
any reason on navigating using useeffect and not in  onSubmit form?

Reply:
It's to check if the user lands on login page somehow and the user is already logged in (since I have saved the info of the user in local storage). If on page load, it finds the user loggedin, it will directly redirect him to the feed page.